core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -624,9 +630,13 @@ private void contextProcessedSuccessfully(SearchContext context) {
     }
 
     private void cleanContext(SearchContext context) {
-        assert context == SearchContext.current();
-        context.clearReleasables(Lifetime.PHASE);
-        SearchContext.removeCurrent();
+        try {
+            assert context == SearchContext.current();
+            context.clearReleasables(Lifetime.PHASE);
+            SearchContext.removeCurrent();
+        } finally {
+            context.decRef();
```

Comment:
Is it expected that the RC after this `decRef` will always be 0?  Can we assert that maybe?  I just want to try to somehow catch future bugs where something does its own `incRef` and fails to later `decRef` and we start leaking `IndexReader`s...